hw02-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,37 @@
+package homework;
+
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    private final TreeMap<Customer, String> customerMap =
```

Comment:
```suggestion
    private final NavigableMap<Customer, String> customerMap =
```

Тип лучше указывать минимально необходимым интерфейсом.